gdal/apps/gdal_grid_lib.cpp

```diff
@@ -823,13 +832,36 @@ GDALDatasetH GDALGrid( const char *pszDest, GDALDatasetH hSrcDataset,
     if ( psOptions->pszSQL )
         nBands++;
 
-    // FIXME
-    int nXSize = psOptions->nXSize;
-    if ( nXSize == 0 )
-        nXSize = 256;
-    int nYSize = psOptions->nYSize;
-    if ( nYSize == 0 )
-        nYSize = 256;
+    int nXSize;
+    int nYSize;
+    if ( psOptions->dfXRes && psOptions->dfYRes )
```

Comment:
To satisfy our pedantic compiler warning rules:
```suggestion
    if ( psOptions->dfXRes != 0 && psOptions->dfYRes != 0 )
```